opm/output/eclipse/Tables.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <ert/ecl/FortIO.hpp>
+#include <ert/ecl/EclKW.hpp>
+#include <ert/ecl/ecl_kw_magic.h>
+
+#include <opm/parser/eclipse/Units/UnitSystem.hpp>
+
+#include <opm/output/eclipse/Tables.hpp>
+
+namespace Opm {
+
+    Tables::Tables( const UnitSystem& units_ ) :
+        units( units_ ),
+        tabdims( "TABDIMS" , TABDIMS_SIZE )
+    {
+    }
+
+
+    void Tables::addData( size_t offset_index, const std::vector<double>& new_data) {
+        this->tabdims[ offset_index ] = this->data.size();
+        std::copy( new_data.begin() , new_data.end() , back_inserter( this->data ));
```

Comment:
This will likely be faster if you use vector's own insert:

`this->data.insert( this->data.end(), new_data.begin(), new_data.end() );`